common/nym-lp/src/keypair.rs

```diff
@@ -0,0 +1,200 @@
+        PublicKey(public_key)
+    }
+}
+
+#[derive(Clone)]
+pub struct PublicKey(SphinxPublicKey);
+
+impl fmt::Debug for PublicKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("PublicKey")
+            .field(&self.to_base58_string())
+            .finish()
+    }
+}
+
+impl Deref for PublicKey {
+    type Target = SphinxPublicKey;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl PublicKey {
+    pub fn to_base58_string(&self) -> String {
+        bs58::encode(self.0.as_bytes()).into_string()
+    }
+
+    pub fn from_base58_string(s: &str) -> Result<Self, LpError> {
+        let bytes: [u8; 32] = bs58::decode(s).into_vec()?.try_into().unwrap();
+        Ok(PublicKey(SphinxPublicKey::from(bytes)))
+    }
+
+    pub fn from_bytes(bytes: &[u8; 32]) -> Result<Self, LpError> {
+        Ok(PublicKey(SphinxPublicKey::from(*bytes)))
+    }
+
+    pub fn as_bytes(&self) -> &[u8; 32] {
+        self.0.as_bytes()
+    }
+}
+
+impl Default for PublicKey {
+    fn default() -> Self {
+        let private_key = PrivateKey::default();
+        private_key.public_key()
+    }
+}
+
+pub struct Keypair {
+    private_key: PrivateKey,
+    public_key: PublicKey,
+}
+
+impl Default for Keypair {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Keypair {
+    pub fn new() -> Self {
+        let private_key = PrivateKey::default();
+        let public_key = private_key.public_key();
+        Self {
+            private_key,
+            public_key,
+        }
+    }
+
+    pub fn from_private_key(private_key: PrivateKey) -> Self {
+        let public_key = private_key.public_key();
+        Self {
+            private_key,
+            public_key,
+        }
+    }
+
+    pub fn from_keys(private_key: PrivateKey, public_key: PublicKey) -> Self {
+        Self {
+            private_key,
+            public_key,
+        }
+    }
+
+    pub fn private_key(&self) -> &PrivateKey {
+        &self.private_key
+    }
+
+    pub fn public_key(&self) -> &PublicKey {
+        &self.public_key
+    }
+}
+
+impl From<KeypairReadable> for Keypair {
+    fn from(keypair: KeypairReadable) -> Self {
+        Self {
+            private_key: PrivateKey::from_base58_string(&keypair.private).unwrap(),
+            public_key: PublicKey::from_base58_string(&keypair.public).unwrap(),
+        }
+    }
+}
+
+impl From<&Keypair> for KeypairReadable {
+    fn from(keypair: &Keypair) -> Self {
+        Self {
+            private: keypair.private_key.to_base58_string(),
+            public: keypair.public_key.to_base58_string(),
+        }
+    }
+}
+impl FromStr for PrivateKey {
+    type Err = LpError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        PrivateKey::from_base58_string(s)
+    }
+}
+
+impl Display for PrivateKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.to_base58_string())
```

Comment:
so we're redacting the content in `Debug` impl but leaving it in `Display`? shouldn't we redact it in both cases?